sklearn/utils/_plotting.py

```diff
@@ -208,13 +218,30 @@ def _validate_curve_kwargs(
         # Ensure `curve_kwargs` is of correct length
         if isinstance(curve_kwargs, Mapping):
             curve_kwargs = [curve_kwargs] * n_curves
+        elif curve_kwargs is None:
+            curve_kwargs = [{}] * n_curves
+
+        if default_curve_kwargs is None:
+            default_curve_kwargs = {}
+        if default_multi_curve_kwargs is None:
+            default_multi_curve_kwargs = {}
+        # This should never happen, as we would not pass the same keys to both
+        # default kwargs
+        common_keys = set(default_curve_kwargs).intersection(default_multi_curve_kwargs)
+        if len(common_keys) > 0:
+            raise ValueError(
+                "`default_curve_kwargs` and `default_multi_curve_kwargs` both contain "
+                f"the keys {common_keys}. The same kwargs cannot be passed to both "
+                "defaults."
+            )
```

Comment:
I don't think that passing the same keys should be a problem. I can imagine a case where you have a slightly different default for multi curves. To me the `default_multi_curve_kwargs` would just override the `default_kurve_kwargs`.

Reply:
Yes I agree. I was being too defensive and didn't think that this is an internal function and will not be used by users. I've amended the functionality.